e2e_tests/tests/cluster/test_checkpoints.py

```diff
@@ -79,7 +80,7 @@ def run_gc_checkpoints_test(checkpoint_storage: Dict[str, str]) -> None:
     for i in range(max_checks):
         time.sleep(1)
```

Comment:
oh snap, is this just a matter of the timeout being too short?

that comment above:

```python
    # Check that the actual checkpoint storage (for shared_fs) reflects the
    # deletions. We want to wait for the GC containers to exit, so check
    # repeatedly with a timeout.
```

would indicate that if checkpoint gc doesn't finish in 30 seconds then we would have this problem.

Can you read through the master logs and find out if this is the case?  I still think it's worth adding better logging to the test case in any case.